var/spack/repos/builtin/packages/root/package.py

```diff
@@ -50,71 +61,432 @@ class Root(CMakePackage):
+            description='Use the specified C++ standard when building.')
+
+    # ###################### Dependencies ##############################
+
+    # minimum cmake version required
     depends_on('cmake@3.4.3:', type='build')
-    depends_on('pkgconfig',   type='build')
+    depends_on('pkgconfig', type='build')
 
-    depends_on('binutils')
-    depends_on('zlib')
-    # depends_on('unuran')
-    depends_on('freetype')
-    depends_on('pcre')
+    depends_on('lz4',    when='@6.13.02:')
+    depends_on('xxhash', when='@6.13.02:')
     depends_on('xz')
-    depends_on('libsm')
-    depends_on('libice')
-    depends_on('libx11')
-    depends_on('libxext')
-    depends_on('libxpm')
-    depends_on('libxft')
-    # depends_on('gif')
+    depends_on('pcre')
+    depends_on('freetype')
     depends_on('libpng')
-    depends_on('jpeg')
-    depends_on('gsl')
-    depends_on('python@2.7:')
-    # depends_on('opengl')
+    depends_on('ncurses')
+    depends_on('zlib')
+
+    # X-Graphics
+    depends_on('libx11',  when="+x")
+    depends_on('libxext', when="+x")
+    depends_on('libxft',  when="+x")
+    depends_on('libxpm',  when="+x")
+
+    # OpenGL
+    depends_on('ftgl',  when="+x+opengl")
+    depends_on('glew',  when="+x+opengl")
+    depends_on('gl',    when="+x+opengl")
+    depends_on('glu',   when="+x+opengl")
+    depends_on('gl2ps', when="+x+opengl")
+
+    # Qt4
+    depends_on('qt', when='+qt4')
+
+    # TODO
+    # Asimage needs one of these two
+    # depends_on('libafterimage',    when='+asimage') - not supported
+    # depends_on('afterstep@2.2.11', when='+asimage') - not supported
+
+    # Optional dependencies
+    depends_on('avahi',     when='+avahi')
+    depends_on('davix',     when='+davix')
+    depends_on('cfitsio',   when='+fits')
+    depends_on('fftw',      when='+fftw')
     depends_on('graphviz', when='+graphviz')
-    # depends_on('kerberos')
-    depends_on('libxml2+python')
-    depends_on('openssl')
+    depends_on('gsl',       when='+gsl')
+    depends_on('http',      when='+http')
+    depends_on('jemalloc',  when='+jemalloc')
+    depends_on('kerberos',  when='+kerberos')
+    depends_on('ldap',      when='+ldap')
+    depends_on('libcxx',    when='+libcxx')
+    # depends_on('minuit',   when='+minuit') - not supported
+    # depends_on('mysql',    when='+mysql')  - not supported
+    depends_on('odbc',      when='+odbc')
+    # depends_on('oracle',   when='+oracle')
+    depends_on('postgresql', when='+postgres')
+    depends_on('pythia@6:6.999',  when='+pythia6')
+    depends_on('pythia@8:8.999',  when='+pythia8')
+    depends_on('python@2.7:',     when='+python', type=('build', 'run'))
+    depends_on('r',         when='+r', type=('build', 'run'))
+    depends_on('r-cpp',     when='+r', type=('build', 'run'))
+    depends_on('r-inside',  when='+r', type=('build', 'run'))
+    depends_on('shadow',    when='+shadow')
+    depends_on('sqlite',    when='+sqlite')
+    depends_on('ssl',       when='+ssl')
```

Comment:
AFAIK, there is currently no such thing as an "ssl" virtual or concrete package in Spack. Did you mean "openssl", or do you want to be generic over SSL implementations (in which case support for that must be added to Spack)?